Sources/BuildSystemIntegration/ExternalBuildSystemAdapter.swift

```diff
@@ -94,10 +94,34 @@ actor ExternalBuildSystemAdapter {
   private var lastRestart: Date?
 
   static package func projectRoot(for workspaceFolder: AbsolutePath, options: SourceKitLSPOptions) -> AbsolutePath? {
-    guard localFileSystem.isFile(workspaceFolder.appending(component: "buildServer.json")) else {
-      return nil
+    var buildServerConfigLocations = [String]()
+
+    #if os(Windows)
+    let workspacePath = workspaceFolder.pathString + "\\.bsp"
+    let localAppDataPath = ProcessInfo.processInfo.environment["LOCALAPPDATA"] ?? "C:\\Users\\Default\\AppData\\Local\\bsp"
+    let programDataPath = ProcessInfo.processInfo.environment["PROGRAMDATA"] ?? "C:\\ProgramData\\bsp"
+
+    buildServerConfigLocations.append(contentsOf: [workspacePath, localAppDataPath, programDataPath])
+    #elseif os(Linux)
+    let workspacePath = workspaceFolder.pathString + "/bsp"
+    let xdgDataHomePath = ProcessInfo.processInfo.environment["XDG_DATA_HOME"] ?? FileManager.default.homeDirectoryForCurrentUser.appendingPathComponent(".local/share/bsp")
+    let xdgDataDirsPath = ProcessInfo.processInfo.environment["XDG_DATA_DIRS"] ?? "/usr/local/share:/usr/share"
+
+    buildServerConfigLocations.append(contentsOf: [workspacePath, xdgDataHomePath, xdgDataDirsPath])
+    #else
+    let workspacePath = workspaceFolder.pathString + "/bsp"
+    let libraryPath = FileManager.default.homeDirectoryForCurrentUser.appending(components: "Library", "Application Support", "bsp").absoluteString
```

Comment:
We don’t need to stitch together the path to `Application Support`. You can use

```swift
FileManager.default.urls(for: .applicationSupportDirectory, in: .userDomainMask)
```

or

```swift
FileManager.default.urls(for: .applicationSupportDirectory, in: .systemDomainMask)
```

instead.

Reply:
Yeah, much better.